envpool/sokoban/sokoban_node.cc

```diff
@@ -184,4 +198,30 @@ bool SokobanNode::GetSuccessors(std::AStarSearch<SokobanNode>* astarsearch,
   return true;
 }
 
+std::pair<int, bool> SokobanNode::SurroundingWalls(
+    const std::pair<int, int>& box) const {
+  int num_walls = 0;
+  bool found_wall = false;
+  bool found_contiguous_wall = false;
+  for (const auto& delta : kDelta) {
+    int new_x = box.first + delta.at(0);
+    int new_y = box.second + delta.at(1);
+    if (CheckWall(new_x, new_y)) {
+      num_walls++;
+      if (found_wall) {
+        found_contiguous_wall = true;
+      }
+      found_wall = true;
+    } else {
+      found_wall = false;
+    }
+  }
+  if (found_wall) {
+    int new_x = box.first + kDelta.at(0).at(0);
+    int new_y = box.second + kDelta.at(0).at(1);
+    found_contiguous_wall = CheckWall(new_x, new_y);
```

Comment:
This should apply the OR operator. `||=` or something. Seems incorrect as it is -- what if it finds a contiguous wall but then there's no wall on the UP part, so it gets set as false?

Reply:
Thanks for pointing this out!